src/runtime/retry.ts

```diff
@@ -0,0 +1,84 @@
+import { Response } from 'node-fetch';
+
+const MAX_REQUEST_RETRY_JITTER = 250;
+const MAX_REQUEST_RETRY_DELAY = 1000;
+const MIN_REQUEST_RETRY_DELAY = 250;
+const DEFAULT_NUMBER_RETRIES = 3;
+const MAX_NUMBER_RETRIES = 10;
+const BASE_DELAY = 250;
+
+/**
+ * @private
+ * Function that returns a random int between a configurable min and max.
+ * @param min The min value
+ * @param max  The max value
+ * @returns {number} The random generated value
+ */
+function getRandomInt(min: number, max: number) {
+  min = Math.ceil(min);
+  max = Math.floor(max);
+  return Math.floor(Math.random() * (max - min) + min); // The maximum is exclusive and the minimum is inclusive
+}
+
+/**
+ * @private
+ * Function that returns a promise which resolves after a configurable amount of milliseconds
+ * @param delay value to be used for the delay
+ * @returns {Promise} A delayed promise
+ */
+async function pause(delay: number) {
+  return new Promise((resolve) => setTimeout(resolve, delay));
+}
+
+export interface RetryConfiguration {
+  /**
+   * Configure the usage of retries.
+   * Defaults to true.
+   */
+  enabled?: boolean;
+  /**
+   * Configure the max amount of retries the SDK should do.
+   * Defaults to 3.
+   */
+  maxRetries?: number;
+  /**
+   * Status Code on which the SDK should trigger retries.
+   * Defaults to 429.
+   */
+  retryWhen?: number;
```

Comment:
Can you make this an array? (eg [link](https://docs.aws.amazon.com/sdkref/latest/guide/feature-retry-behavior.html#:~:text=Transient%20errors%20(HTTP%20status%20codes%20400%2C%20408%2C%20500%2C%20502%2C%20503%2C%20and%20504)%20and%20throttling%20errors%20(HTTP%20status%20codes%20400%2C%20403%2C%20429%2C%20502%2C%20503%2C%20and%20509)%20can%20all%20potentially%20be%20retried.))

Reply:
Done